File: Torneig_Mort_Spring/src/main/java/org/eljust/DTO/JugadorDTO.java
package org.eljust.DTO;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

/**
 * Aquest DTO de jugador el farem servir per a mostrar tota la informació
 * relllevant d'un jugador,
 */
@Getter
@Setter
public class JugadorDTO {

	private Long idJugador;
	private String nomJugador;
	private int edat;
	private boolean sancionat;
	private List<EquipSimpleDTO> elsEquips = new ArrayList<>();

}

File: app_torneig_flutter/lib/models/jugador_dto.dart
import 'package:app_torneig/models/equip_simple_dto.dart';

class JugadorDTO {
  int? id;
  String? nom;
  int? edat;
  bool? esSancionat;
  List<EquipSimpleDTO>? elsEquips;

  JugadorDTO({
    required this.id,
    this.nom,
    this.edat,
    this.elsEquips,
    this.esSancionat,
  });

  JugadorDTO.fromJSON(Map<String, dynamic> objecteJSON) {
    id = objecteJSON['idJugador'];
    nom = objecteJSON['nomJugador'] ?? "";
    edat = objecteJSON['edat'];
    esSancionat = objecteJSON['sancionat'];
    elsEquips = objecteJSON['elsEquips'] != null
        ? List<EquipSimpleDTO>.from(objecteJSON['elsEquips']
            .map((equip) => EquipSimpleDTO.fromJSON(equip)))
        : null;
  }
}
